src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryBasedAsyncOverrides.cs

```diff
@@ -0,0 +1,225 @@
+        {
+            var compilation = context.Compilation;
+
+            if (!TryGetRequiredSymbols(compilation, out RequiredSymbols symbols))
+                return;
+
+            context.RegisterSymbolAction(AnalyzeNamedType, SymbolKind.NamedType);
+
+            return;
+
+            // Local functions.
+            void AnalyzeNamedType(SymbolAnalysisContext context)
+            {
+                var type = (INamedTypeSymbol)context.Symbol;
+
+                //  We only report a diagnostic if the type directly subclasses stream. We don't report diagnostics
+                //  if there are any bases in the middle.
+
+                if (!symbols.StreamType.Equals(type.BaseType, SymbolEqualityComparer.Default))
+                    return;
+
+                //  We use 'FirstOrDefault' because there could be multiple overrides for the same method if
+                //  there are compiler errors.
+
+                IMethodSymbol? readAsyncArrayOverride = GetOverridingMethodSymbols(type, symbols.ReadAsyncArrayMethod).FirstOrDefault();
+                IMethodSymbol? readAsyncMemoryOverride = GetOverridingMethodSymbols(type, symbols.ReadAsyncMemoryMethod).FirstOrDefault();
+                IMethodSymbol? writeAsyncArrayOverride = GetOverridingMethodSymbols(type, symbols.WriteAsyncArrayMethod).FirstOrDefault();
+                IMethodSymbol? writeAsyncMemoryOverride = GetOverridingMethodSymbols(type, symbols.WriteAsyncMemoryMethod).FirstOrDefault();
+
+                //  For both ReadAsync and WriteAsync, if the array-based form is overridden and the memory-based
+                //  form is not, we report a diagnostic. We report separate diagnostics for ReadAsync and WriteAsync. 
+
+                if (readAsyncArrayOverride is not null && readAsyncMemoryOverride is null)
+                {
+                    var diagnostic = CreateDiagnostic(type, readAsyncArrayOverride, symbols.ReadAsyncMemoryMethod);
+                    context.ReportDiagnostic(diagnostic);
+                }
+
+                if (writeAsyncArrayOverride is not null && writeAsyncMemoryOverride is null)
+                {
+                    var diagnostic = CreateDiagnostic(type, writeAsyncArrayOverride, symbols.WriteAsyncMemoryMethod);
+                    context.ReportDiagnostic(diagnostic);
+                }
+            }
+
+            static Diagnostic CreateDiagnostic(INamedTypeSymbol violatingType, IMethodSymbol arrayBasedOverride, IMethodSymbol memoryBasedMethod)
+            {
+                RoslynDebug.Assert(arrayBasedOverride.OverriddenMethod is not null);
+
+                //  We want to underline the name of the violating type in the class declaration. If the violating type
+                //  is a partial class, we underline all partial declarations.
+
+                return violatingType.CreateDiagnostic(
+                    Rule,
+                    violatingType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
```

Comment:
We normally use just the type/method `.Name` in diagnostic messages. `MinimallyQualifiedFormat` actually produces a qualified name, which lead to very long names.

Reply:
@mavasani Fixed. I've changed the message so it makes sense with simple names (the message was originally worded with the assumption that the arguments of each overload would be displayed).